src/x509.c

```diff
@@ -5859,19 +5863,20 @@ static int X509PrintSignature(WOLFSSL_BIO* bio, WOLFSSL_X509* x509, int indent)
             ret = WOLFSSL_FAILURE;
     }
 
-    if (ret == WOLFSSL_SUCCESS) {
-        sig = (unsigned char*)XMALLOC(sigSz, NULL, DYNAMIC_TYPE_TMP_BUFFER);
-        if (sig == NULL)
-            ret = WOLFSSL_FAILURE;
-    }
+    /* Leave function if the desired content to print
+     * is only the signature algorithm */
+    if (algOnly) {
+        if (obj != NULL)
+            wolfSSL_ASN1_OBJECT_free(obj);
 
-    if (ret == WOLFSSL_SUCCESS) {
-        if (wolfSSL_X509_get_signature(x509, sig, &sigSz) <= 0)
-            ret = WOLFSSL_FAILURE;
+        if (sig != NULL)
+            XFREE(sig, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+
+        return ret;
     }
 
     if (ret == WOLFSSL_SUCCESS) {
-        if ((tmpLen = XSNPRINTF(tmp, sizeof(tmp), "%*s", indent + 4, ""))
+        if ((tmpLen = XSNPRINTF(tmp, sizeof(tmp), "%*s", indent + 5, ""))
```

Comment:
is the increased indent only for CRL cases or was this off by 1 even with X509 print?

Reply:
It was off by one for X509 as well